clients/src/main/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImpl.java

```diff
@@ -531,7 +537,13 @@ private CompletableFuture<Void> invokeOnPartitionsRevokedOrLostToReleaseAssignme
      * request is sent out with it.
      */
     private void transitionToSendingLeaveGroup() {
-        memberEpoch = ConsumerGroupHeartbeatRequest.LEAVE_GROUP_MEMBER_EPOCH;
+        if (state == MemberState.FATAL) {
+            log.warn("Member won't send leave group request because it is in FATAL state");
```

Comment:
nit: `Member {} with epoch {}`? I am not sure whether the epoch makes any sense here.

Reply:
Added them both. I'm also not sure that the epoch will be very helpful in this terminal state, but I say we add it, it won't hurt and it might help to understand the evolution with previous logs.